### PR TITLE
Plfm 4612

### PR DESCRIPTION
**Pull request by brucehoff**

### CODE REVIEW

File: lib/jdomodels/src/main/resources/schema/evaluation/Evaluation-ddl.sql [hunk 1]
```diff
@@ -10,6 +10,8 @@ CREATE TABLE IF NOT EXISTS JDOEVALUATION (
     SUBMISSION_INSTRUCTIONS_MESSAGE mediumblob,
     SUBMISSION_RECEIPT_MESSAGE mediumblob,
     QUOTA mediumblob,
+    START_TIMESTAMP bigint(20),
+	END_TIMESTAMP bigint(20),
```
Comment (jay-hodgson) [inline lib/jdomodels/src/main/resources/schema/evaluation/Evaluation-ddl.sql]: tab or spaces

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dbo/EvaluationDBO.java [hunk 2]
```diff
@@ -328,7 +355,12 @@ public boolean equals(Object obj) {
 			@Override
 			public EvaluationDBO createDatabaseObjectFromBackup(
 					EvaluationBackup backup) {
-				return EvaluationTranslationUtil.createDatabaseObjectFromBackup(backup);
+				EvaluationDBO dbo =  EvaluationTranslationUtil.createDatabaseObjectFromBackup(backup);
+				// fill out start and end time stamps
+				Evaluation dto = new Evaluation();
```
Comment (jay-hodgson) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dbo/EvaluationDBO.java:360]: todo: remove after stack release (please open a new jira ticket!)

**Comment by jay-hodgson** [comment]
Please test EvaluationDBO

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java [hunk 3]
```diff
@@ -177,7 +175,7 @@ public Evaluation get(String id) throws DatastoreException, NotFoundException {
 
 
 	@Override
-	public List<Evaluation> getAccessibleEvaluationsForProject(String projectId, List<Long> principalIds, ACCESS_TYPE accessType, long limit, long offset) 
+	public List<Evaluation> getAccessibleEvaluationsForProject(String projectId, List<Long> principalIds, ACCESS_TYPE accessType, Long now, long limit, long offset)
```
Comment (jay-hodgson) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java]: Please rename "now" to a param name that suggests that it's applying a time filter.

File: services/repository-managers/src/test/java/org/sagebionetworks/evaluation/manager/EvaluationManagerTest.java [hunk 4]
```diff
@@ -120,56 +119,78 @@ public void setUp() throws DatastoreException, NotFoundException, InvalidModelEx
...
 	
+	@Test
+	public void testGetEvaluationByContentSourceActiveOnly() throws Exception {
+
+    	evaluations= Collections.singletonList(evalWithId);
+    	when(mockEvaluationDAO.getAccessibleEvaluationsForProject(eq(EVALUATION_CONTENT_SOURCE), (List<Long>)any(), eq(ACCESS_TYPE.READ), anyLong(), anyLong(), anyLong())).
+    		thenReturn(Collections.EMPTY_LIST);
+
+		List<Evaluation> qr = evaluationManager.getEvaluationByContentSource(ownerInfo, EVALUATION_CONTENT_SOURCE, true, 10L, 0L);
+		assertTrue(qr.isEmpty());
```
Comment (jay-hodgson) [inline services/repository-managers/src/test/java/org/sagebionetworks/evaluation/manager/EvaluationManagerTest.java:172]: Please verify getAccessibleEvaluationsForProject is called (with non-null time filter in this case).

File: services/repository-managers/src/test/java/org/sagebionetworks/evaluation/manager/EvaluationManagerTest.java [hunk 5]
```diff
@@ -120,56 +119,78 @@ public void setUp() throws DatastoreException, NotFoundException, InvalidModelEx
...
+	}
+	
 	@Test
 	public void testUpdateEvaluationAsOwner() throws DatastoreException, InvalidModelException, ConflictingUpdateException, NotFoundException, UnauthorizedException {
+    	when(mockEvaluationDAO.get(eq(EVALUATION_ID))).thenReturn(evalWithId);
```
Comment (jay-hodgson) [inline services/repository-managers/src/test/java/org/sagebionetworks/evaluation/manager/EvaluationManagerTest.java]: clean up indentation (whitespace)

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/EvaluationController.java [hunk 6]
```diff
@@ -165,19 +162,30 @@ Evaluation getEvaluation(
 	 * </p>
 	 * 
 	 * @param id - the ID of the Project.
+	 * @param activeOnly - if 'true' then include only those evaluations with rounds defined and for which the current time is in one of the rounds
```
Comment (jay-hodgson) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/EvaluationController.java]: dashes and capitalization.

Comment (jay-hodgson) [reply services/repository/src/main/java/org/sagebionetworks/repo/web/controller/EvaluationController.java]: if true then _return_ ...

File: services/repository/src/test/java/org/sagebionetworks/repo/web/service/EvaluationServiceTest.java [hunk 7]
```diff
@@ -102,7 +102,7 @@ public void testGetAllSubmissions() {
 		List<Submission> expectedRes = new LinkedList<Submission>();
 		when(mockSubmissionManager.getAllSubmissions(userInfo, evalId,  SubmissionStatusEnum.OPEN, limit, offset)).thenReturn(expectedRes);
 		// Call under test
-		evaluationService.getAllSubmissions(userId, evalId, SubmissionStatusEnum.OPEN, limit, offset, null);
+		evaluationService.getAllSubmissions(userId, evalId, SubmissionStatusEnum.OPEN, limit, offset);
```
Comment (jay-hodgson) [inline services/repository/src/test/java/org/sagebionetworks/repo/web/service/EvaluationServiceTest.java:133]: please backfill basic test for getEvaluationsInRange and getEvaluationByContentSource

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dbo/EvaluationDBO.java (hunk 2 above)
Comment (brucehoff) [reply lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dbo/EvaluationDBO.java:360]: https://sagebionetworks.jira.com/browse/PLFM-5940

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**